packages/app/src/pages/Specs/Runner.vue

```diff
@@ -48,20 +50,76 @@ subscription SpecPageContainer_specsChange {
 }
 `
 
-useSubscription({ query: SpecPageContainer_SpecsChangeDocument })
+gql`
+subscription Runner_ConfigChange {
+  configChange {
+    id
+    serveConfig
+  }
+}
+`
 
 const isRunMode = window.__CYPRESS_MODE__ === 'run'
 
+// subscriptions are used to trigger live updates without
+// reloading the page.
+// this is only useful in open mode - in run mode, we don't
+// use GraphQL, so we pause the
+// subscriptions so they never execute.
+const shouldPauseSubscriptions = isRunMode && window.top === window
+
+useSubscription({
+  query: SpecPageContainer_SpecsChangeDocument,
+  pause: shouldPauseSubscriptions,
+})
+
 // in run mode, we are not using GraphQL or urql
 // for performance - run mode does not need the
 // same level of runner interactivity as open mode.
 // by setting `pause: true` in run mode, urql will not trigger any
 // requests, which is what we want.
 const query = useQuery({
   query: SpecPageContainerDocument,
-  pause: isRunMode && window.top === window,
+  pause: shouldPauseSubscriptions,
 })
 
+let initialLoad = true
+
+const specStore = useSpecStore()
+
+// When cypress.config.js is changed,
+// we respond by updating the runner with the latest config
+// and re-running the current spec with the new config values.
+const configChangeHandler: SubscriptionHandlerArg<any, any> = (
+  _prev: Runner_ConfigChangeSubscription | undefined,
+  next: Runner_ConfigChangeSubscription,
+) => {
+  if (!next.configChange?.serveConfig) {
+    throw Error(`Did not get expected serveConfig from subscription`)
+  }
+
+  if (!initialLoad && specStore.activeSpec) {
+    try {
```

Comment:
Not super ideal, since we don't know that eventManager will absolute exist here (potential race condition, you changed `cypress.config.js` while the spec is still "Spec is loading...", so I just did a try/catch.